scripts/linux/cross-platform-rust-build.sh

```diff
@@ -206,39 +296,29 @@ case "$PACKAGE_OS.$PACKAGE_ARCH" in
             echo \'[target.aarch64-unknown-linux-gnu]\' > ~/.cargo/config &&
             echo \'linker = \"aarch64-linux-gnu-gcc\"\' >> ~/.cargo/config &&
             export AARCH64_UNKNOWN_LINUX_GNU_OPENSSL_LIB_DIR=/usr/lib/aarch64-linux-gnu &&
-            export AARCH64_UNKNOWN_LINUX_GNU_OPENSSL_INCLUDE_DIR=/usr/include &&
+            export AARCH64_UNKNOWN_LINUX_GNU_OPENSSL_INCLUDE_DIR=/usr/include && 
+            cd /project/$BUILD_PATH &&
+            echo \'Installing rustup\' &&
+            curl -sSLf https://sh.rustup.rs | sh -s -- -y &&
+            . ~/.cargo/env &&
         '
+        MAKE_FLAGS="'CARGOFLAGS=$CARGOFLAGS --target aarch64-unknown-linux-gnu'"
+        MAKE_FLAGS="$MAKE_FLAGS 'TARGET=target/aarch64-unknown-linux-gnu/release'"
+        MAKE_FLAGS="$MAKE_FLAGS 'STRIP_COMMAND=/usr/aarch64-linux-gnu/bin/strip'"
         ;;
 esac
 
+if [ -n "$RUST_TARGET" ]; then
+    RUST_TARGET_COMMAND="rustup target add $RUST_TARGET &&"
+fi
+
 if [ -z "$SETUP_COMMAND" ]; then
     echo "Unrecognized target [$PACKAGE_OS.$PACKAGE_ARCH]" >&2
     exit 1
 fi
 
-case "$PACKAGE_OS" in
-    *)
-        case "$PACKAGE_ARCH" in
-            amd64)
-                MAKE_FLAGS="'CARGOFLAGS=$CARGOFLAGS --target x86_64-unknown-linux-musl'"
-                MAKE_FLAGS="$MAKE_FLAGS 'TARGET=target/x86_64-unknown-linux-musl/release'"
-                MAKE_FLAGS="$MAKE_FLAGS 'STRIP_COMMAND=strip'"
-                ;;
-            arm32v7)
-                MAKE_FLAGS="'CARGOFLAGS=$CARGOFLAGS --target armv7-unknown-linux-gnueabihf'"
-                MAKE_FLAGS="$MAKE_FLAGS 'TARGET=target/armv7-unknown-linux-gnueabihf/release'"
-                MAKE_FLAGS="$MAKE_FLAGS 'STRIP_COMMAND=/usr/arm-linux-gnueabihf/bin/strip'"
-                ;;
-            aarch64)
-                MAKE_FLAGS="'CARGOFLAGS=$CARGOFLAGS --target aarch64-unknown-linux-gnu'"
-                MAKE_FLAGS="$MAKE_FLAGS 'TARGET=target/aarch64-unknown-linux-gnu/release'"
-                MAKE_FLAGS="$MAKE_FLAGS 'STRIP_COMMAND=/usr/aarch64-linux-gnu/bin/strip'"
-                ;;
-        esac
+MAKE_COMMAND="make release $MAKE_FLAGS"
 
-        MAKE_COMMAND="make release $MAKE_FLAGS"
-        ;;
-esac
```

Comment:
Moving everything into one block might make things messy going forward, but I suppose it is fine for now.

Reply:
Possibly yes. For now we only have alpine, but it would not be too hard to break it down in smaller files.